pkg/app/middleware_test.go

```diff
@@ -0,0 +1,58 @@
+package app
+
+import (
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/openshift/assisted-service/pkg/thread"
+	"github.com/sirupsen/logrus"
+)
+
+func TestMiddleWare(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Middleware test")
+}
+
+var _ = Describe("Middleware test", func() {
+
+	l := logrus.New()
+	l.SetOutput(ioutil.Discard)
+
+	It("Healthcheck failure test", func() {
+		timeout := 20*time.Millisecond
+		mHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {})
+		failure := thread.New(l, "failed test", 100*time.Millisecond, func() {
+			time.Sleep(10 * time.Millisecond)
```

Comment:
why do you need a sleep here if the thread is working with a ticker?

Reply:
no need